### ISSUE TITLE
Create adaptive Icon for android 8

### ISSUE THREAD

**Issue by adrian-bl** [vanilla-music/vanilla#660]
Documentation is at:
https://developer.android.com/guide/practices/ui_guidelines/icon_design_adaptive.html

**Comment by Adonai** [comment]
... why?

**Comment by adrian-bl** [comment]
Is there any launcher actually using them?

**Comment by Adonai** [comment]
No

**Comment by adrian-bl** [comment]
Reopening this: My Phone just got an Upgrade to 8.1 which increases my motivation for an adoptive icon:

![image](https://user-images.githubusercontent.com/568734/39662500-5cef1c0a-5062-11e8-9186-96208c39042e.png)

@andiandi13 as you created the current icon: Would be glad to get some feedback/ideas on how such an icon would look like. I'm also not that much fixed on our current 'loudspeaker'-theme - so using something completely different as an icon would be fine.

**Comment by adrian-bl** [comment]
Closing as dupe of #660

**Comment by andiandi13** [comment]
@adrian-bl Sorry I never seen your message.

But I think you juste created an icon force the next update or is it a test ?

**Comment by andiandi13** [comment]
Why is your launcher keeping the full icon inside the circle ? Oreo do that ?

I'm still on KitKat and I tried an Oreo launcher, it's quite nice if it's like this imo :

![]( https://image.noelshack.com/fichiers/2018/25/7/1529860637-screenshot-2018-06-24-19-16-18.png )

**Comment by adrian-bl** [comment]
> Oreo do that ?

Yes: Applications compiled with newer SDK (26) versions will (forcefully) be opted-in to adaptive icons: The default of my launcher is to create a round adaptive icon.

Also see: https://developer.android.com/guide/practices/ui_guidelines/icon_design_adaptive

**Comment by adrian-bl** [comment]
Oh: you are running an (old) nightly build, right?

That was my first attempt in commit 1b2776319456fc0a9e1d5f0b05c1190912fd822e - i didn't really like it and went back to the original icon.

The latest version looks like this:

![image](https://user-images.githubusercontent.com/568734/41821726-eb9a4f2a-77e4-11e8-80a0-81267f4df191.png)

**Comment by andiandi13** [comment]
No I'm running the last play store version actually.

I'll make some attempts when I'll have some spare time.

But tell me, can't we directly make a new round icon ?
Players like Musicolet are quite nice like this.

**Comment by adrian-bl** [comment]
Oh? Shamecube for me then: looks like i compiled 1.61 for the play store at the wrong commit id then :-)
The 1.61 tag (which should have been used) does not include the adaptive icon.

> But tell me, can't we directly make a new round icon ?

The problem is that adaptive icons are not always round: A launcher may choose to display the icon as a square :-/ (Only the inner 72x72dp of the foreground layer are guaranteed to be displayed, the background layer can have different shapes)

**Comment by andiandi13** [comment]
No I don't think you made a mistake, as I said, I'm on kit Kat, its just an Oreo launcher that behaves like that.

> The problem is that adaptive icons are not always round: A launcher may choose to display the icon as a square :-/

Hmm I see so we _must_ create a squared icon (with round corners or no, whatever...) ?

**Comment by adrian-bl** [comment]
> Hmm I see so we must create a squared icon (with round corners or no, whatever...) ?

We just need to create a logo which fits in the 72x72dp foreground layer (See link to developers.android.com above).

The launcher can then choose to render the 108x108dp background layer in any shape it likes.

**Comment by andiandi13** [comment]
OK I see, the link is very clear.

So if I create an svg icon, is a 72x72 mask with the fullscreen icon sufficient ? If yes do I have to set a background color behind the icon to fill the corners ?

**Comment by adrian-bl** [comment]
> So if I create an svg icon, is a 72x72 mask with the fullscreen icon sufficient 

The background color could be transparent: We could then set the background layer to a solid color (android automatically stacks both layers - and the foreground layer is free to use transparency)

Btw: here is another nice article about adaptive icons with more examples and details: https://medium.com/google-design/designing-adaptive-icons-515af294c783

**Comment by andiandi13** [comment]
Alright ill check that.
But if we make a transparent background, can it be a solid color also in firmwares <8.0 with a custom shape ?

**Comment by adrian-bl** [comment]
Yes: launchers without adaptive icon support will use a completely different drawable (icon.png)

**Comment by andiandi13** [comment]
OK thank you

**Comment by andiandi13** [comment]
@adrian-bl I juste saw a logo proposal for Vanilla on a website : https://steemit.com/utopian-io/@lowlevel/logo-design-proposal-for-vanilla-music-app?sort=new

Its pretty good and unique imo what do you think ?

**Comment by adrian-bl** [comment]
It is unique, but i'm not a big fan of it :-/ (I agree that the non-gradient version looks much better however).

**Comment by andiandi13** [comment]
Ok. I tried to think of an unique icon but tbh it's quite hard to stand out from what you see everywhere.

I'll continue to draw some draft though
